zpay32/invoice_internal_test.go

```diff
@@ -230,13 +296,510 @@ func TestParseTimestamp(t *testing.T) {
 	for i, test := range tests {
 		time, err := parseTimestamp(test.data)
 		if (err == nil) != test.valid {
-			t.Errorf("Data decoding test %d failed: %v", i, err)
+			t.Errorf("timestamp decoding test %d failed: %v", i, err)
 			return
 		}
 		if test.valid && time != test.result {
-			t.Errorf("Timestamp decoding test %d failed: expected "+
-				"timestamp %d, got %d", i, test.result, time)
+			t.Errorf("%d) failed decoding timestamp: "+
+				"expected %d, got %d",
+				i, test.result, time)
+			return
+		}
+	}
+}
+
+// TestParseFieldDataLength checks that the 16 bit length is properly parsed.
+func TestParseFieldDataLength(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		data   []byte
+		valid  bool
+		result uint16
+	}{
+		{
+			data:  []byte{},
+			valid: false, // empty data
+		},
+		{
+			data:  []byte{0x0},
+			valid: false, // data too short
+		},
+		{
+			data:  []byte{0x0, 0x0, 0x0},
+			valid: false, // data too long
+		},
+		{
+			// The first byte is <= 3 bits long.
+			data:   []byte{0x1, 0x2},
+			valid:  true,
+			result: 34,
+		},
+		{
+			// The first byte is > 3 bits long.
+			data:   []byte{0xa, 0x0},
```

Comment:
Awesome! Could you also include cases for the two extremes? (`{0x0, 0x0}` and `{0b11111, 0x11111}` I think)